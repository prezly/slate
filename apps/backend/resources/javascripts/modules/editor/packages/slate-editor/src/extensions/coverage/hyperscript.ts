/* eslint-disable @typescript-eslint/no-namespace */

import type { CoverageEntry } from '@prezly/sdk';
import { COVERAGE_NODE_TYPE, PARAGRAPH_NODE_TYPE } from '@prezly/slate-types';
import type { ReactNode } from 'react';
import { createHyperscript, createText } from 'slate-hyperscript';

declare global {
    namespace JSX {
        interface IntrinsicElements {
            'h-coverage': {
                children?: ReactNode;
                coverage: {
                    id: CoverageEntry['id'];
                };
                uuid: string;
            };
            'h-p': {
                children?: ReactNode;
            };
        }
    }
}

export const hyperscript = createHyperscript({
    elements: {
        'h-coverage': { type: COVERAGE_NODE_TYPE },
        'h-p': { type: PARAGRAPH_NODE_TYPE },
    },
    creators: {
        'h-text': createText,
    },
});
