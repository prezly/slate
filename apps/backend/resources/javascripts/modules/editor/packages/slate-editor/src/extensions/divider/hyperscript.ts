/* eslint-disable @typescript-eslint/no-namespace */

import { DIVIDER_NODE_TYPE, PARAGRAPH_NODE_TYPE } from '@prezly/slate-types';
import type { ReactNode } from 'react';
import { createHyperscript, createText } from 'slate-hyperscript';

declare global {
    namespace JSX {
        interface IntrinsicElements {
            'h-divider': {
                children?: ReactNode;
            };
            'h-p': {
                children?: ReactNode;
            };
        }
    }
}

export const hyperscript = createHyperscript({
    elements: {
        'h-divider': { type: DIVIDER_NODE_TYPE },
        'h-p': { type: PARAGRAPH_NODE_TYPE },
    },
    creators: {
        'h-text': createText,
    },
});
