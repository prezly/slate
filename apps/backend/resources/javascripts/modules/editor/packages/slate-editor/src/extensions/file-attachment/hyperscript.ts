/* eslint-disable @typescript-eslint/no-namespace */

import { ATTACHMENT_NODE_TYPE, PARAGRAPH_NODE_TYPE } from '@prezly/slate-types';
import type { UploadedFile } from '@prezly/uploadcare';
import type { ReactNode } from 'react';
import { createHyperscript, createText } from 'slate-hyperscript';

declare global {
    namespace JSX {
        interface IntrinsicElements {
            'h-file-attachment': {
                children?: ReactNode;
                description: string;
                file: UploadedFile;
            };
            'h-p': {
                children?: ReactNode;
            };
        }
    }
}

export const hyperscript = createHyperscript({
    elements: {
        'h-file-attachment': { type: ATTACHMENT_NODE_TYPE },
        'h-p': { type: PARAGRAPH_NODE_TYPE },
    },
    creators: {
        'h-text': createText,
    },
});
