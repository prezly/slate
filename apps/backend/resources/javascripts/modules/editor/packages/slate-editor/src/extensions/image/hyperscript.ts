/* eslint-disable @typescript-eslint/no-namespace */

import type { ImageLayout } from '@prezly/slate-types';
import { IMAGE_NODE_TYPE, PARAGRAPH_NODE_TYPE } from '@prezly/slate-types';
import type { UploadedFile } from '@prezly/uploadcare';
import type { ReactNode } from 'react';
import { createHyperscript, createText } from 'slate-hyperscript';

declare global {
    namespace JSX {
        interface IntrinsicElements {
            'h-image': {
                children?: ReactNode;
                file: UploadedFile;
                href: string;
                layout: ImageLayout;
                width: string;
            };
            'h-p': {
                children?: ReactNode;
            };
        }
    }
}

export const hyperscript = createHyperscript({
    elements: {
        'h-image': { type: IMAGE_NODE_TYPE },
        'h-p': { type: PARAGRAPH_NODE_TYPE },
    },
    creators: {
        'h-text': createText,
    },
});
