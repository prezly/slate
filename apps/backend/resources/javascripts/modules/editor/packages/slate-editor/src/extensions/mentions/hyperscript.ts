/* eslint-disable @typescript-eslint/no-namespace */

import type { VariableNode } from '@prezly/slate-types';
import { PARAGRAPH_NODE_TYPE } from '@prezly/slate-types';
import { VARIABLE_NODE_TYPE } from '@prezly/slate-types';
import type { ReactNode } from 'react';
import { createHyperscript, createText } from 'slate-hyperscript';

declare global {
    namespace JSX {
        interface IntrinsicElements {
            'h-placeholder-mention': {
                children?: ReactNode;
                key: VariableNode['key'];
            };
            'h-p': {
                children?: ReactNode;
            };
        }
    }
}

export const hyperscript = createHyperscript({
    elements: {
        'h-placeholder-mention': { type: VARIABLE_NODE_TYPE },
        'h-p': { type: PARAGRAPH_NODE_TYPE },
    },
    creators: {
        'h-text': createText,
    },
});
