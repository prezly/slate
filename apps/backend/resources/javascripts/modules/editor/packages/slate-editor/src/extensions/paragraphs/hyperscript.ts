/* eslint-disable @typescript-eslint/no-namespace */

import { PARAGRAPH_NODE_TYPE } from '@prezly/slate-types';
import type { ReactNode } from 'react';
import { createHyperscript, createText } from 'slate-hyperscript';

declare global {
    namespace JSX {
        interface IntrinsicElements {
            'h-p': {
                children?: ReactNode;
            };
        }
    }
}

export const hyperscript = createHyperscript({
    elements: {
        'h-p': { type: PARAGRAPH_NODE_TYPE },
    },
    creators: {
        'h-text': createText,
    },
});
