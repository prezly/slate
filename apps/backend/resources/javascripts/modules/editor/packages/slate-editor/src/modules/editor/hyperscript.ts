/* eslint-disable @typescript-eslint/no-namespace */

import {
    BULLETED_LIST_NODE_TYPE,
    DIVIDER_NODE_TYPE,
    LIST_ITEM_NODE_TYPE,
    LIST_ITEM_TEXT_NODE_TYPE,
    NUMBERED_LIST_NODE_TYPE,
    PARAGRAPH_NODE_TYPE,
    QUOTE_NODE_TYPE,
} from '@prezly/slate-types';
import type { ReactNode } from 'react';
import { createHyperscript, createText } from 'slate-hyperscript';

declare global {
    namespace JSX {
        interface IntrinsicElements {
            'h-divider': {
                children?: ReactNode;
            };
            'h-li': {
                children?: ReactNode;
            };
            'h-li-text': {
                children?: ReactNode;
            };
            'h-ol': {
                children?: ReactNode;
            };
            'h-p': {
                children?: ReactNode;
            };
            'h-ul': {
                children?: ReactNode;
            };
            'h:blockquote': {
                children?: ReactNode;
            };
            'h:text': {
                children?: ReactNode;
            };
        }
    }
}

export const hyperscript = createHyperscript({
    elements: {
        'h-p': { type: PARAGRAPH_NODE_TYPE },
        'h-divider': { type: DIVIDER_NODE_TYPE },
        'h-ol': { type: NUMBERED_LIST_NODE_TYPE },
        'h-ul': { type: BULLETED_LIST_NODE_TYPE },
        'h-li': { type: LIST_ITEM_NODE_TYPE },
        'h-li-text': { type: LIST_ITEM_TEXT_NODE_TYPE },
        'h:blockquote': { type: QUOTE_NODE_TYPE },
    },
    creators: {
        'h-text': createText,
        'h:text': createText,
    },
});
