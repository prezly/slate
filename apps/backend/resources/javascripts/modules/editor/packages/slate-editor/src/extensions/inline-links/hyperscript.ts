/* eslint-disable @typescript-eslint/no-namespace,@typescript-eslint/no-unused-vars */
import { LINK_NODE_TYPE, PARAGRAPH_NODE_TYPE } from '@prezly/slate-types';
import { createEditor as createSlateEditor } from 'slate';
import { createHyperscript, createEditor as createEditorFactory } from 'slate-hyperscript';

import { createEditor } from '#modules/editor';

import { InlineLinksExtension } from './InlineLinksExtension';

// This allows tests to include Slate Nodes written in JSX without TypeScript complaining.
declare global {
    namespace JSX {
        interface IntrinsicElements {
            [elemName: string]: any; // eslint-disable-line
        }
    }
}

const extensions = [InlineLinksExtension()];

export const hyperscript = createHyperscript({
    elements: {
        link: { type: LINK_NODE_TYPE },
        paragraph: { type: PARAGRAPH_NODE_TYPE },
    },
    creators: {
        editor: createEditorFactory(() => createEditor(createSlateEditor(), () => extensions)),
    },
});
