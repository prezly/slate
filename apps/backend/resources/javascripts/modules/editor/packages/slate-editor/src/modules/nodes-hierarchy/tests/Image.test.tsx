/** @jsx hyperscript */

import { Alignment, ImageLayout } from '@prezly/slate-types';
import { Editor } from 'slate';

import { hyperscript } from '../../../hyperscript';

describe('nodes-hierarchy / Image', () => {
    it('should be kept after normalization', function () {
        const editor = (
            <editor>
                <h:image
                    align={Alignment.CENTER}
                    href=""
                    layout={ImageLayout.CONTAINED}
                    new_tab
                    width="100%"
                    file={{
                        version: 2,
                        uuid: 'c849afb1-57cc-48c2-ab31-10b5e1555a33',
                        filename:
                            'png-transparent-pink-cross-stroke-ink-brush-pen-red-ink-brush-ink-leave-the-material-text.png',
                        mime_type: 'image/png',
                        size: 41346,
                        original_width: 920,
                        original_height: 900,
                        effects: [],
                    }}
                >
                    <h:text>Caption</h:text>
                </h:image>
            </editor>
        ) as unknown as Editor;

        const expected = (
            <editor>
                <h:image
                    align={Alignment.CENTER}
                    href=""
                    layout={ImageLayout.CONTAINED}
                    new_tab
                    width="100%"
                    file={{
                        version: 2,
                        uuid: 'c849afb1-57cc-48c2-ab31-10b5e1555a33',
                        filename:
                            'png-transparent-pink-cross-stroke-ink-brush-pen-red-ink-brush-ink-leave-the-material-text.png',
                        mime_type: 'image/png',
                        size: 41346,
                        original_width: 920,
                        original_height: 900,
                        effects: [],
                    }}
                >
                    <h:text>Caption</h:text>
                </h:image>
            </editor>
        ) as unknown as Editor;

        Editor.normalize(editor, { force: true });

        expect(editor.children).toMatchSnapshot();
        expect(editor.children).toEqual(expected.children);
        expect(editor.selection).toEqual(expected.selection);
    });
});
