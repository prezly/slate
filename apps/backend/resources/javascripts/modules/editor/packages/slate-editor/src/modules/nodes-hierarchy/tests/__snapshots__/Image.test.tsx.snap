// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`nodes-hierarchy / Image should be kept after normalization 1`] = `
[
  {
    "align": "center",
    "children": [
      {
        "text": "Caption",
      },
    ],
    "file": {
      "effects": [],
      "filename": "png-transparent-pink-cross-stroke-ink-brush-pen-red-ink-brush-ink-leave-the-material-text.png",
      "mime_type": "image/png",
      "original_height": 900,
      "original_width": 920,
      "size": 41346,
      "uuid": "c849afb1-57cc-48c2-ab31-10b5e1555a33",
      "version": 2,
    },
    "href": "",
    "layout": "contained",
    "new_tab": true,
    "type": "image-block",
    "width": "100%",
  },
]
`;
